pkg/apis/pipeline/v1beta1/pipeline_validation_test.go

```diff
@@ -315,7 +378,10 @@ func TestPipelineSpec_Validate_Failure(t *testing.T) {
 				},
 			}},
 		},
+		expectedError: *apis.ErrGeneric(`invalid value: couldn't add link between foo and bar: task foo depends on bar but bar wasn't present in Pipeline`, "tasks").Also(
+			apis.ErrInvalidValue("expected resource great-resource to be from task bar, but task bar doesn't exist", "tasks[1].resources.inputs[0].from")),
```

Comment:
This test is problematic as the `dag.Build` cyclic error is not deterministic… need to find a way to fix that test